### PR TITLE
Add layout design for current weather

### PR DESCRIPTION
**Pull request by minaahilimtiaz**
Add recyclerview in home fragment

Add cardview and constraintlayout for design

Add adapter and viewholder classes

### CODE REVIEW

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.example.android.weatherapp.R
+import com.example.android.weatherapp.adapter.HomeScreenAdapter
+import com.example.android.weatherapp.databinding.FragmentHomeScreenBinding
+import com.example.android.weatherapp.models.CurrentWeatherModel
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Remove extra code

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [HomeScreenFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class HomeScreenFragment : Fragment() {
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Create base class for fragment

File: app/src/main/res/layout/current_weather_layout.xml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            android:layout_height="match_parent">
+
+            <androidx.constraintlayout.widget.Guideline
+                android:id="@+id/guideline"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:orientation="horizontal"
+                app:layout_constraintGuide_begin="2dp" />
+
+            <TextView
```
Comment (noumanahmad001) [inline app/src/main/res/layout/current_weather_layout.xml]: Too much properties there, try to move these to styles

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        /**
+         * Use this factory method to create a new instance of
+         * this fragment using the provided parameters.
+         *
+         * @param param1 Parameter 1.
+         * @param param2 Parameter 2.
+         * @return A new instance of fragment HomeScreenFragment.
+         */
+        // TODO: Rename and change types and number of parameters
+        @JvmStatic
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: If you are not using any param then please remove the ARG_PARAMs

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        val binding: FragmentHomeScreenBinding = DataBindingUtil.inflate(
+            inflater, R.layout.fragment_home_screen, container, false)
+
+
+        var list= listOf<CurrentWeatherModel>(
+            CurrentWeatherModel("Very Sunny", "17C",
+                "Abottabad, Pakistan", "Friday, August 18, 2020"))
+
+        val adapter= HomeScreenAdapter()
+        binding.homeScreenRcv.adapter=adapter
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Rename this homeScreenRcv to more meaningful name . 
Add space before and after '='

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        adapter.data= list
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: there should be space before =

File: app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.android.weatherapp.R
+import com.example.android.weatherapp.models.CurrentWeatherModel
+import kotlinx.android.synthetic.main.current_weather_layout.view.*
+
+class HomeScreenAdapter:RecyclerView.Adapter<HomeScreenAdapter.ViewHolder>() {
+
+    var data = listOf<CurrentWeatherModel>()
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt:13]: variable data should be renamed.

File: app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeScreenAdapter.ViewHolder {
+        return ViewHolder.inflateAndGetViewHolder(parent)
+    }
+
+    override fun onBindViewHolder(holder: HomeScreenAdapter.ViewHolder, position: Int) {
+        val item = data.get(position)
+        holder.bindingViewToData(item)
+    }
+
+    class ViewHolder (itemView: View): RecyclerView.ViewHolder(itemView) {
```
Comment (noumanahmad001) [inline app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt]: ViewHolder class should be renamed.

File: app/src/main/java/com/example/android/weatherapp/models/CurrentWeatherModel.kt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.example.android.weatherapp.models
+
+import android.service.controls.templates.TemperatureControlTemplate
```
Comment (AbdulMateenChughtai) [inline app/src/main/java/com/example/android/weatherapp/models/CurrentWeatherModel.kt]: What is this import supposed to do?

File: app/src/main/res/layout/current_weather_layout.xml [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+                android:layout_marginEnd="8dp"
+                android:layout_marginBottom="8dp"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toEndOf="@id/barrier2"
+                app:layout_constraintTop_toBottomOf="@id/guideline"
+                app:layout_constraintVertical_bias="0.5"
+                app:srcCompat="@drawable/sunny_image" />
+
+            <androidx.constraintlayout.widget.Barrier
```
Comment (AbdulMateenChughtai) [inline app/src/main/res/layout/current_weather_layout.xml]: For now, it's okay, but you can achieve almost this layout by using just one guideline. Always try to keep xml files as lean as possible.

File: app/src/main/java/com/example/android/weatherapp/models/CurrentWeatherModel.kt (hunk 9 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/models/CurrentWeatherModel.kt]: It is an unused import added while adding **temperature** variable. I have removed it.

File: app/src/main/res/layout/current_weather_layout.xml (hunk 10 above)
Comment (minaahilimtiaz) [reply app/src/main/res/layout/current_weather_layout.xml]: Should i leave it for now or change it?

Comment (AbdulMateenChughtai) [reply app/src/main/res/layout/current_weather_layout.xml]: Keep it as is for now.

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -4,15 +4,12 @@
...
 
-    <TextView
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+    <fragment
```
Comment (AbdulMateenChughtai) [inline app/src/main/res/layout/activity_main.xml:9]: Create fragments dynamically/programmatically.

**Review by AbdulMateenChughtai — APPROVED** [review]

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt (hunk 1 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Removed the auto-generated code from HomeScreenFragment Class

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt (hunk 2 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Added base class for fragment and activity

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt (hunk 5 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Renamed it to ScreenRecyclerview and added space before and '='

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt (hunk 6 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Added space before and after '='

File: app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt (hunk 8 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt]: Renamed it to **CurrentWeatherViewHolder**

File: app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt (hunk 7 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/adapter/HomeScreenAdapter.kt:13]: renamed it to **currentWeatherList**

File: app/src/main/res/layout/current_weather_layout.xml (hunk 3 above)
Comment (minaahilimtiaz) [reply app/src/main/res/layout/current_weather_layout.xml]: Added style for cardview and textview

File: app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt (hunk 4 above)
Comment (minaahilimtiaz) [reply app/src/main/java/com/example/android/weatherapp/fragment/HomeScreenFragment.kt]: Yes not using it. Removed it.

**Pull request merged**